### DISCUSSION TITLE
How to setup styles for App using string?

### DISCUSSION THREAD

**Discussion by laixintao** [Textualize/textual#3243] [Q&A] [answered]
I have a tcss string from somewhere in my application, like this:

```
foo {
  width: 100%;
}
```

I want to set it to my App, however, it seems that, for now, textual can only support it from a filepath.

In order to do that, I have to make a file using the content, then let my `App` open and read it again.

```
def make_style_file():
        hashname = hashlib.md5(content.encode()).hexdigest()

        file = CACHE_LOCATION / f"{hashname}.tcss"

        if file.exists():
            logger.info("%s already exists, use cached one", file)
            return file

        file.parents[0].mkdir(parents=True, exist_ok=True)
        with open(file, "w") as f:
            f.write(content)
            logger.info("Cache tss file: %s", file)

        return file
```

Is there a better way doing this? thanks.

**Comment by davep** [comment, marked as answer]
I'm a little confused about your question, especially given the code you include, but I *think* you're asking if you can specify CSS inline in your code, rather than having to write it in an external file; is that correct?

If so, have you read [the guide to App basics](https://textual.textualize.io/guide/app/), especially [the section about CSS](https://textual.textualize.io/guide/app/#css), and most specifically [the part that describes how to do inline CSS](https://textual.textualize.io/guide/app/#classvar-css)?

**Reply by laixintao** [reply]
thank you @davep , and I am sorry I opened an issue instead of discussion.

I have read this guide (and documentations are so perfect!)

However, what I want is that, let's say I want my users are able to config styles by themself, with those styles there are other configs. I'd rather let all of the configs stays in one file instead of app-config.yaml + app-style.tcss.

Now with your answer I realized that the `CSS` is on `class` instead of object, so with this, I can implement this way:

```python
MyApp.CSS = style_content
app = MyApp()
app.run()
```

(I think if we have a method that can set the style on App instead of modifying the class before creating the object would be better.)

**Reply by davep** [reply]
Thanks for the clarification, that makes more sense now. Yes, your solution there is a good workaround for now. In the future I think we'd like to add a public API for loading more CSS from the application instance ([this is something I'd like to add for the benefit of Frogmouth](https://github.com/Textualize/frogmouth/issues/42), for example).
